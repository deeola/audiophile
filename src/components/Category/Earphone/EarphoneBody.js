import React from "react";
import ProductWithImage from "../shared/ProductWithImage";
import { EarphonCont } from "./Earphonestyling";
import earphonedesktop from "../../../assets/category-earphones/desktop/earphoneDesktop.jpg";

import earphonetablet from "../../../assets/category-earphones/tablet/image-yx1-earphones.jpg";
import earphonemobile from "../../../assets/category-earphones/mobile/image-yx1-earphones.jpg";

const EarphoneBody = () => {
  // use inner width to determine image soure
  let earphoneimage = earphonedesktop;

  if (window.innerWidth < 802) {
    earphoneimage = earphonetablet;
  } else if (window.innerWidth < 490) {
    earphoneimage = earphonemobile;
  }
  return (
    <EarphonCont data-testid='earphone'>
      <ProductWithImage
        link="/yx1"
        img={earphoneimage}
        label="SEE PRODUCT"
        np={"NEW PRODUCT"}
        title="YX1 wireless earphones"
        subtitle={
          "Tailor your listening experience with bespoke dynamic drivers from the new YX1 Wireless Earphones. Enjoy incredible high-fidelity sound even in noisy environments with its active noise cancellation feature."
        }
      />
    </EarphonCont>
  );
};

export default EarphoneBody;
