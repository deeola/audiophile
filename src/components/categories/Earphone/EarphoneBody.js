import React from 'react'
import PropTypes from 'prop-types'
import EarSection from './EarSection'

function EarphoneBody(props) {
    return (
        <div>
            <EarSection />
        </div>
    )
}


export default EarphoneBody

