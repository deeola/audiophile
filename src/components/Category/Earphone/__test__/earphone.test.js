import { render, cleanup } from "@testing-library/react";
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";
import { BrowserRouter } from "react-router-dom";

import EarphoneBody from "../EarphoneBody";



afterEach(cleanup);

it("renders earphone section correctly", () => {
  const section = document.createElement("section");
  ReactDOM.render(
    <BrowserRouter>
      <EarphoneBody />
    </BrowserRouter>,
    section
  );
});

it("contains one element children", () => {
  const { getByTestId } = render(
    <BrowserRouter>
      <EarphoneBody />
    </BrowserRouter>
  );

  expect(getByTestId('earphone').childElementCount).toBe(1)
});

// create snapshot

it('matches snapshot', () => {
    const tree  = renderer.create(
        <BrowserRouter>
        <EarphoneBody />
      </BrowserRouter>
    ).toJSON()

    expect(tree).toMatchSnapshot()
})