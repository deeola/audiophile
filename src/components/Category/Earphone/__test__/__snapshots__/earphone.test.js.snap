// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<section
  className="sc-papXJ bxEehH"
  data-testid="earphone"
>
  <div
    className="sc-bczRLJ gIwRId"
    data-testid="productimage"
  >
    <img
      alt="YX1 wireless earphones"
      className="sc-dkzDqf gRVFrI"
      src="earphoneDesktop.jpg"
    />
    <article
      className="sc-hKMtZM bQLcuJ"
      data-testid="productbutton"
    >
      <div
        className="sc-eCYdqJ bMkDvt"
      >
        <p
          className="sc-iBkjds dIztKO"
          data-testid="np"
        >
           
          NEW PRODUCT
        </p>
        <h2
          className="sc-jSMfEi imCpJA"
          data-testid="title"
        >
          YX1 wireless earphones
        </h2>
        <p
          className="sc-gKXOVf dgecyH"
          data-testid="subtitle"
        >
          Tailor your listening experience with bespoke dynamic drivers from the new YX1 Wireless Earphones. Enjoy incredible high-fidelity sound even in noisy environments with its active noise cancellation feature.
        </p>
      </div>
      <a
        className="sc-ftvSup KzLBC"
        color="primary"
        data-testid="productbutton-YX1 wireless earphones"
        hover="primary"
        href="/yx1"
        onClick={[Function]}
      >
        SEE PRODUCT
      </a>
    </article>
  </div>
</section>
`;
